libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/puck/LocationPuckOptions.kt

```diff
@@ -0,0 +1,191 @@
+     * @return the [Builder] that created the [LocationPuckOptions]
+     */
+    fun toBuilder(): Builder = Builder(context).apply {
+        freeDrivePuck(freeDrivePuck)
+        destinationPreviewPuck(destinationPreviewPuck)
+        routePreviewPuck(routePreviewPuck)
+        activeNavigationPuck(activeNavigationPuck)
+        arrivalPuck(arrivalPuck)
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as LocationPuckOptions
+
+        if (context != other.context) return false
+        if (freeDrivePuck != other.freeDrivePuck) return false
+        if (destinationPreviewPuck != other.destinationPreviewPuck) return false
+        if (routePreviewPuck != other.routePreviewPuck) return false
+        if (activeNavigationPuck != other.activeNavigationPuck) return false
+        if (arrivalPuck != other.arrivalPuck) return false
+
+        return true
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun hashCode(): Int {
+        var result = context.hashCode()
+        result = 31 * result + freeDrivePuck.hashCode()
+        result = 31 * result + destinationPreviewPuck.hashCode()
+        result = 31 * result + routePreviewPuck.hashCode()
+        result = 31 * result + activeNavigationPuck.hashCode()
+        result = 31 * result + arrivalPuck.hashCode()
+        return result
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun toString(): String {
+        return "LocationPuckOptions(" +
+            "context=$context, " +
+            "freeDrivePuck=$freeDrivePuck, " +
+            "destinationPreviewPuck=$destinationPreviewPuck, " +
+            "routePreviewPuck=$routePreviewPuck, " +
+            "activeNavigationPuck=$activeNavigationPuck, " +
+            "arrivalPuck=$arrivalPuck" +
+            ")"
+    }
+
+    /**
+     * Builder of [LocationPuckOptions]
+     */
+    class Builder(private val context: Context) {
+
+        private var freeDrivePuck = regularPuck(context)
+        private var destinationPreviewPuck = regularPuck(context)
+        private var routePreviewPuck = regularPuck(context)
+        private var activeNavigationPuck = navigationPuck(context)
+        private var arrivalPuck = navigationPuck(context)
+
+        /**
+         * Apply the same [LocationPuck2D] or [LocationPuck3D] to location puck in all different
+         * navigation states
+         * @param defaultPuck [LocationPuck] to be used in all navigation states
+         */
+        fun defaultPuck(defaultPuck: LocationPuck): Builder = apply {
+            this.freeDrivePuck = defaultPuck
+            this.destinationPreviewPuck = defaultPuck
+            this.routePreviewPuck = defaultPuck
+            this.activeNavigationPuck = defaultPuck
+            this.arrivalPuck = defaultPuck
+        }
```

Comment:
A small suggestion:
```kotlin
class Builder(private val context: Context) {

    private var freeDrivePuck: LocationPuck? = null
    private var destinationPreviewPuck: LocationPuck? = null
    private var routePreviewPuck: LocationPuck? = null
    private var activeNavigationPuck: LocationPuck? = null
    private var arrivalPuck: LocationPuck? = null
    private var defaultPuck: LocationPuck? = null

    fun defaultPuck(defaultPuck: LocationPuck): Builder = apply {
        this.defaultPuck = defaultPuck
    }

    ...

    fun build(): LocationPuckOptions {
        val regularPuck by lazy { regularPuck(context) }
        val navigationPuck by lazy { navigationPuck(context) }
        return LocationPuckOptions(
            context,
            freeDrivePuck ?: defaultPuck ?: regularPuck,
            destinationPreviewPuck ?: defaultPuck ?: regularPuck,
            routePreviewPuck ?: defaultPuck ?: regularPuck,
            activeNavigationPuck ?: defaultPuck ?: navigationPuck,
            arrivalPuck ?: defaultPuck ?: navigationPuck
        )
    }
    ...
}
```
 to allow `defaultPuck` to be called anywhere in the builder chain.

```kotlin
val opt1 = LocationPuckOptions.Builder(context)
   .freeDrivePuck(myFDPuck)
   .defaultPuck(myPuck)
   .build()

val opt2 = LocationPuckOptions.Builder(context)
   .defaultPuck(myPuck)
   .freeDrivePuck(myFDPuck)
   .build()

assertEquals(opt1, opt2) 
```

Reply:
Done.